pkg/ccl/changefeedccl/scheduled_changefeed.go

```diff
@@ -0,0 +1,854 @@
+	}
+
+	sj.SetScheduleDetails(details)
+
+	var args changefeedpb.ScheduledChangefeedExecutionArgs
+	args.ScheduledChangefeedStatement = tree.AsStringWithFlags(changefeedNode, tree.FmtParsable|tree.FmtShowPasswords)
+	any, err := pbtypes.MarshalAny(&args)
+	if err != nil {
+		return nil, err
+	}
+	sj.SetExecutionDetails(
+		tree.ScheduledChangefeedExecutor.InternalName(), jobspb.ExecutionArguments{Args: any},
+	)
+
+	return sj, nil
+}
+
+// dryRunChangefeed executes Changefeed in dry-run mode: we simply execute Changefeed
+// under transaction savepoint, and then rollback to that save point.
+func dryRunCreateChangefeed(
+	ctx context.Context,
+	p sql.PlanHookState,
+	scheduleID int64,
+	createChangefeedNode *tree.CreateChangefeed,
+) error {
+	sp, err := p.ExtendedEvalContext().Txn.CreateSavepoint(ctx)
+	if err != nil {
+		return err
+	}
+
+	err = func() error {
+		annotated := &annotatedChangefeedStatement{
+			CreateChangefeed: createChangefeedNode,
+			CreatedByInfo: &jobs.CreatedByInfo{
+				Name: jobs.CreatedByScheduledJobs,
+				ID:   scheduleID,
+			},
+		}
+		changefeedFn, err := planCreateChangefeed(ctx, p, annotated)
+		if err != nil {
+			return err
+		}
+		return invokeCreateChangefeed(ctx, changefeedFn)
+	}()
+
+	if rollbackErr := p.ExtendedEvalContext().Txn.RollbackToSavepoint(ctx, sp); rollbackErr != nil {
+		return rollbackErr
+	}
+	return err
+}
+
+func planCreateChangefeed(
+	ctx context.Context, p sql.PlanHookState, createChangefeedStmt tree.Statement,
+) (sql.PlanHookRowFn, error) {
+	fn, cols, _, _, err := changefeedPlanHook(ctx, createChangefeedStmt, p)
+
+	if err != nil {
+		return nil, errors.Wrapf(err, "changefeed table eval: %q", tree.AsString(createChangefeedStmt))
+	}
+	if fn == nil {
+		return nil, errors.Newf("changefeed table eval: %q", tree.AsString(createChangefeedStmt))
+	}
+	if len(cols) != 1 {
+		return nil, errors.Newf("unexpected result columns")
+	}
+	return fn, nil
+}
+
+func invokeCreateChangefeed(ctx context.Context, createChangefeedFn sql.PlanHookRowFn) error {
+	resultCh := make(chan tree.Datums) // No need to close
+	g := ctxgroup.WithContext(ctx)
+
+	g.GoCtx(func(ctx context.Context) error {
+		select {
+		case <-resultCh:
+			return nil
+		case <-ctx.Done():
+			return ctx.Err()
+		}
+	})
+
+	g.GoCtx(func(ctx context.Context) error {
+		return createChangefeedFn(ctx, nil, resultCh)
+	})
+
+	return g.Wait()
+}
+
+func makeScheduleDetails(opts map[string]string) (jobspb.ScheduleDetails, error) {
+	var details jobspb.ScheduleDetails
+	if v, ok := opts[optOnExecFailure]; ok {
+		if err := schedulebase.ParseOnError(v, &details); err != nil {
+			return details, err
+		}
+	}
+
+	if v, ok := opts[optOnPreviousRunning]; ok {
+		if err := schedulebase.ParseWaitBehavior(v, &details); err != nil {
+			return details, err
+		}
+	}
+	return details, nil
+}
+
+// getRedactedChangefeedNode returns a copy of the argument `changefeed`, but with all
+// the secret information redacted.
+func getRedactedChangefeedNode(
```

Comment:
can you double check if we can't use some of the existing functions (e.g. changefeedJobDescription) instead? (I'm not saying you can or that you should switch, just want to make sure)